integration/unplugin/src/index.ts

```diff
@@ -458,7 +458,11 @@ export const rawPlugin: Parameters<typeof createUnplugin<PluginOptions>>[0] =
 
         if (implicitExtension) {
           config.resolve ??= {};
-          config.resolve.extensions ??= [];
+          config.resolve.extensions ??= [
+            // Vite's DEFAULT_EXTENSIONs from
+            // https://github.com/vitejs/vite/blob/main/packages/vite/src/node/constants.ts#L29
+            '.mjs', '.js', '.mts', '.ts', '.jsx', '.tsx', '.json',
```

Comment:
Could we import this from Vite if present or would it be too much of a hassle?

Reply:
It's not exported from Vite. I was thinking of submitting a Vite PR to expose it.

But there's also an issue of adding Vite as a dependency. Is there a nice way to copy the constant over with just a dev dependency? I guess I could do it in a build script without much trouble.

Reply:
I succeeded in getting this to work via `import { DEFAULT_EXTENSIONS } from '../../../node_modules/vite/dist/node/constants.js'`, but sadly esbuild wouldn't treeshake that module, so it ended up adding a bunch of unnecessary code to the bundle.

To fix this, I've added a Node step to `build.sh` that extracts this to an also-checked-in file. Does that seem reasonable?

Reply:
Yes, that seems like a good solution.

Reply:
Another idea I had: we could put this in a `prepare` script so it runs when getting dependencies but not when building, which would slightly speed up the dev cycle (I yarn build more often than yarn install). Not sure the time penalty is enough to bother though.